Lib/test/test_typing.py

```diff
@@ -3692,10 +3692,15 @@ def test_c_functions(self):
 
 
 class NewTypeTests:
+    def cleanup(self):
```

Comment:
Wow this is some great investigation @uriyyo! I forgot that the global `typing` and the `py_typing` used for `NewType` tests are different module objects. I was quite puzzled for a while about why the `BaseTestCase` clearing didn't work :).

Reply:
@Fidget-Spinner I spent a lot of time investigating C code because though that problem was at it. It was tricky but interesting to investigate this memory leak)